lib

```diff
@@ -22,6 +22,7 @@ fi
 
 die() {
     echo "$1" >&2
+    clean
```

Comment:
I think lib shouldn't  have a dependency on a function that is not defined in the file. That break the whole idea behind a generic library that can be sourced.